ksql-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java

```diff
@@ -1088,7 +1101,7 @@ public Node visitRegisterType(final RegisterTypeContext context) {
     }
 
     private void throwOnUnknownNameOrAlias(final SourceName name) {
-      if (!sources.contains(name)) {
+      if (sources.isPresent() && !sources.get().contains(name)) {
```

Comment:
Makes me think we should pull such validation out of this builder and add as a secondary step to the KsqlParser.

Reply:
Yeah this makes sense to me - not in scope for this patch though.